goyaml.v3/encode_test.go

```diff
@@ -731,6 +731,40 @@ func (s *S) TestSortedOutput(c *C) {
 	}
 }
 
+func (s *S) TestQuotingMaintained(c *C) {
+	var buf bytes.Buffer
+	var yamlValue map[string]interface{}
+	const originalYaml = `data:
+    A1: "0x0000000000000000000000010000000000000000"
+    A2: "0x000000000000000000000000FFFFFFFFFFFFFFFF"
```

Comment:
maintaining the quoting on A1/A2 is good ... do we also know that YAMLToJSON works properly here? that's what will be used when reading a yaml manifest and sending it to the kube-apiserver

Reply:
I'll add a test case for YAMLToJSON

Reply:
@liggitt I added some test cases to the `TestYAMLtoJSON` function in `yaml_test.go` in my last commit 742f5e6, which I believe covers both of these scenarios since it first converts the YAML to JSON, and then converts it back, but please let me know if I misunderstood!